pkg/util/validation/validate.go

```diff
@@ -1,10 +1,39 @@
 package validation
```

Comment:
I assume this is what was pulled in from Cortex?  I don't disagree with the decision as you've been far deeper in this than I, but instead of doing this could we make `AddEntry` for the `CodedErrorSampler` accept any `error` and just assume 500 if it's not a `CodedError`?